ddtrace/llmobs/_llmobs.py

```diff
@@ -75,30 +92,127 @@ def _stop_service(self) -> None:
+        dd_llmobs_no_apm: bool = False,
+        dd_site: Optional[str] = None,
+        dd_api_key: Optional[str] = None,
+        dd_env: Optional[str] = None,
+        dd_service: Optional[str] = None,
+        tracer=None,
+    ):
+        """
+        Enable LLM Observability tracing.
+
+        :param str ml_app: The name of your ml application.
+        :param List[str] integrations: A list of integrations to enable auto-tracing for.
+        :param bool dd_llmobs_no_apm: Set to `true` to disable sending non-LLM Observability data to Datadog.
+        :param str dd_site: Your datadog site.
+        :param str dd_api_key: Your datadog api key.
+        :param str dd_env: Your environment name.
+        :param str dd_service: Your service name.
+        """
         if cls._instance is not None:
             log.debug("%s already enabled", cls.__name__)
             return
 
+        if os.getenv("DD_LLMOBS_ENABLED") and not asbool(os.getenv("DD_LLMOBS_ENABLED")):
+            LLMObs._toggle(False)
+            log.debug("LLMObs.enable() called when DD_LLMOBS_ENABLED is set to false, not starting LLMObs service")
+            return
+
+        # grab required values for LLMObs
+        # reading from environment variables is needed in case preload script wasn't called.
+        config._dd_site = config._dd_site or os.getenv("DD_SITE") or dd_site
+        config._dd_api_key = config._dd_api_key or os.getenv("DD_API_KEY") or dd_api_key
+        config._llmobs_ml_app = config._llmobs_ml_app or os.getenv("DD_LLMOBS_APP_NAME") or ml_app
+
+        # validate required values for LLMObs
         if not config._dd_api_key:
-            cls.enabled = False
+            LLMObs._toggle(False)
             raise ValueError(
                 "DD_API_KEY is required for sending LLMObs data. "
                 "Ensure this configuration is set before running your application."
             )
+        if not config._dd_site:
+            LLMObs._toggle(False)
+            raise ValueError(
+                "DD_SITE is required for sending LLMObs data. "
+                "Ensure this configuration is set before running your application."
+            )
         if not config._llmobs_ml_app:
-            cls.enabled = False
+            LLMObs._toggle(False)
             raise ValueError(
                 "DD_LLMOBS_APP_NAME is required for sending LLMObs data. "
                 "Ensure this configuration is set before running your application."
             )
 
+        # grab optional values
+        config.env = os.getenv("DD_ENV") or config.env or dd_env
+        config.service = os.getenv("DD_SERVICE") or config.service or dd_service
+
+        if dd_llmobs_no_apm or asbool(os.getenv("DD_LLMOBS_NO_APM", "false")):
+            for k, v in cls._no_apm_env_config.items():
+                # don't override environment variables are are explicitly set.
+                if not os.getenv(k):
+                    os.environ[k] = v
```

Comment:
additionally, whether we're modifying internal config options, or we end up going with modifying env vars, it might be helpful to emit a log saying something like
```
'dd_llmobs_no_apm' was set to True, setting DD_OPENAI_METRICS_ENABLED to False
```
etc., just so it's clear to a user with debug logs turned on or a support engineer that this is what happened under the hood. wdyt?

Reply:
i like the debug log idea, and agree with not changing the remote config env var

unfortunately `DD_OPENAI_METRICS_ENABLED` isn't set at the config object level, the openai integration directly reads from that env var

Reply:
ahh ok, i think this is fine then